File: apps/rails/spec/factories/contractor_profiles.rb
# frozen_string_literal: true

FactoryBot.define do
  factory :contractor_profile do
    association :user, factory: [:user, :contractor]
    available_for_hire { true }
    available_hours_per_week { rand(1..35) }
    description { Faker::Quote.famous_last_words }
  end
end

File: e2e/factories/contractorProfiles.ts
import { faker } from "@faker-js/faker";
import { db } from "@test/db";
import { usersFactory } from "@test/factories/users";
import { contractorProfiles } from "@/db/schema";

export const contractorProfilesFactory = {
  create: async (overrides: Partial<typeof contractorProfiles.$inferInsert> = {}) => {
    const [contractorProfile] = await db
      .insert(contractorProfiles)
      .values({
        userId: overrides.userId || (await usersFactory.create()).user.id,
        availableForHire: true,
        availableHoursPerWeek: faker.number.int({ min: 1, max: 35 }),
        description: faker.lorem.sentence(),
        ...overrides,
      })
      .returning();

    return { contractorProfile };
  },
};
